app/models/incident.rb

```diff
@@ -95,4 +105,19 @@ def serializable_hash(options = nil)
 
     json
   end
+
+  def visible_to?(user)
+    case visibility
+    when "visible"
+      true # Everyone can see public incidents
+    when "staff"
+      user&.staff? || user&.can_manage_incidents? # Staff and WRC can see staff incidents
+    when "draft"
+      user&.can_manage_incidents? # Only WRC can see drafts
+    end
+  end
```

Comment:
Interestingly, the checks you're doing here are different from the controller (see my other, longer comment about `can_view_delegate_matters`) but I feel like they _should_ be the same.

Perhaps a common check method can be reused both here and in the controller?